core/src/ru/spbau/blackout/abilities/gravity/GravityAbility.java

```diff
@@ -0,0 +1,59 @@
+package ru.spbau.blackout.abilities.gravity;
+
+import ru.spbau.blackout.abilities.Ability;
+import ru.spbau.blackout.abilities.simpleshell.SimpleShellAbility;
+import ru.spbau.blackout.entities.Character;
+
+public class GravityAbility extends SimpleShellAbility {
+
+    static final String CAST_SOUND_PATH = "sounds/fire.ogg";
+    static final String EFFECT_PATH = "effects/gravity/gravity.pfx";;
+    static final String ICON_PATH = "abilities/gravity/icon.png";
+    static final String NAME = "Gravity";
+    static final float MAX_CHARGE_TIME = 4f;
+    static final float SHELL_MASS = 10000;
+    static final float DISTANCE = 12f;
+    static final float SHELL_START_SPEED = 15;
+    static final float TIME_TO_LIVE = DISTANCE / SHELL_START_SPEED;
+    static final float MAX_DAMAGE_BASE = 0.3f;
+    static final float MAX_DAMAGE_PER_LEVEL = 0.12f;
+    static final float MAX_FORCE_BASE = 750;
+    static final float MAX_FORCE_PER_LEVEL = 300;
+    static final float RADIUS = 7;
+
+    protected GravityAbility(Definition def, Character character) {
+        super(def, character);
+    }
+
+    public static class Definition extends SimpleShellAbility.Definition {
+
+        public Definition(int level) {
+            super(level, new GravityObject.Definition());
+        }
+
+        @Override
+        public void updateLevel() {
+            super.updateLevel();
+            ((GravityObject.Definition) shellDef).maxForce = MAX_FORCE_BASE + MAX_FORCE_PER_LEVEL * (getLevel() - 1);
```

Comment:
Yes, there is a problem with definitions. Each `GameObject` class has its `Definition` class. Also, each `GameObject` created from corresponding `Definition` object and should be able to refer to it. So, I do this:

```
protected GameObject(Definition def, Uid uid, float x, float y) {
    this.def = def;
    ...
}

public final Definition getDef() {
    return def;
}
```

But, since this code is inside the base class, when I want to get a definition, I have to do unsafe type cast.
Unfortunately, I don't see any possible ways to mitigate this besides to keep many references to the same definition (on each layer of inheritance).
I don't think that double dispatch could help.